source/Missions/Services/Arena/CoopArenaController.cs

```diff
@@ -190,58 +187,170 @@ private void Handle_JoinInfo(MessagePayload<NetworkMissionJoinInfo> payload)
 
             Logger.Information("Spawning {EntityType} called {AgentName}({AgentID}) from {Peer} with {ControlledAgentCount} controlled agents",
                 joinInfo.CharacterObject.IsPlayerCharacter ? "Player" : "Agent",
-                joinInfo.CharacterObject.Name, newAgentId, 
+                joinInfo.CharacterObject.Name, newAgentId,
                 netPeer.EndPoint,
                 joinInfo.UnitIdString?.Length);
 
-            if(joinInfo.IsPlayerAlive)
+            if (joinInfo.IsPlayerAlive)
             {
-                Agent newAgent = SpawnAgent(startingPos, joinInfo.CharacterObject, true);
+                Agent newAgent = SpawnAgent(startingPos, joinInfo.CharacterObject, true, joinInfo.Equipment);
                 _agentRegistry.RegisterNetworkControlledAgent(netPeer, joinInfo.PlayerId, newAgent);
             }
 
             for (int i = 0; i < joinInfo.UnitIdString?.Length; i++)
             {
                 Agent tempAi = SpawnAgent(joinInfo.UnitStartingPosition[i], CharacterObject.Find(joinInfo.UnitIdString[i]), true);
-                
+
                 _agentRegistry.RegisterNetworkControlledAgent(netPeer, joinInfo.UnitId[i], tempAi);
             }
         }
 
         public override void OnAgentShootMissile(Agent shooterAgent, EquipmentIndex weaponIndex, Vec3 position, Vec3 velocity, Mat3 orientation, bool hasRigidBody, int forcedMissileIndex)
```

Comment:
Instead of this, create postfix events for `AddMissileSingleUsageAux` and `AddMissileAux`.
Create handlers for the respective messages created by those.
Send network events if the agent is controlled. Call the same methods on the client and add that missile to the `_missiles` list similar to `OnAgentShootMissile`.

I will also update the issue with these needed updates.